physics/module_sf_noahmplsm.F90

```diff
@@ -5382,160 +5501,432 @@ subroutine sfcdif3(parameters,iloc    ,jloc    ,iter    ,sfctmp  ,qair    ,ur
     tem2   = max(fveg, 0.1_kind_phys)
     zvfun1 = sqrt(tem1 * tem2)
     gdx    = sqrt(garea1)
+    
+    gdx    = 3000.0   ! this will remove gdx effect
+    zvfun1 = 1.0      ! this will remove zvfun effect
 
     if(thsfc_loc) then            ! Use local potential temperature
       tvs   = tgb * virtfac
     else                          ! Use potential temperature referenced to 1000 hPa
       tvs   = tgb/prsik1x * virtfac
     endif
 
-    call stability (zlvlb, zvfun1, gdx, tv1, thv1, ur, z0m, z0h, tvs, grav, thsfc_loc,  &
+    call gfs_stability (zlvlb, zvfun1, gdx, tv1, thv1, ur, z0m, z0h, tvs, grav, thsfc_loc,  &
          rb1, fm,fh,fm10,fh2,cm,ch,stress1,fv)
 
   end subroutine sfcdif3
 
+!== begin gfs_stability ==================================================================================
+
+subroutine gfs_stability                                              &
+!  ---  inputs:
+          ( z1, zvfun, gdx, tv1, thv1, wind, z0max, ztmax, tvs, grav,  &
+            thsfc_loc,                                                 &
+!  ---  outputs:
+            rb, fm, fh, fm10, fh2, cm, ch, stress, ustar)
+
+! Documentation below refers to UTN and STN which are:
+!  UTN (Unstable Tech Note) : NCEP Office Note 356
+!  STN (Stable Tech Note)   : NCEP Office Note 321
+
+integer, parameter :: kp = kind_phys
```

Comment:
Why are you using two parameterized precisions in this routine, kp and kind_phys? 
I see kp is set to kind_phys, so it's not causing any problems, but having two is confusing. My suggestion would be to use one.